commons/all/src/main/java/org/infinispan/commons/configuration/io/ConfigurationReader.java

```diff
@@ -43,22 +43,30 @@ public Builder withProperties(Properties properties) {
       }
 
       public Builder withType(MediaType type) {
-         this.type = type;
+         if (type != null) {
```

Comment:
I personally think just using Objects.requireNull is better. A user shouldn't ever supply null for these right?

Reply:
Right. Repushed